src/fixtures/ide_fixture.py

```diff
@@ -91,3 +96,53 @@ def setup_intellij(intellij_config, config):
 
     intellij.close_ide()
     time.sleep(5)
+
+
+@pytest.fixture(scope="function")
+def configurations_object(config, intellij_config, vscode_config, app_name, analysis_data, ide):
+    # region construct configuration object and fill it from the data json
+
+    application_config = intellij_config if ide == "intellij" else vscode_config
+
+    configurations = ConfigurationsObject()
+
+    application_data = analysis_data[app_name]
+    model_json_path = f"{application_config['plugin_cache_path']}/model.json"
+
+    uuid = generate_uuid()
+
+    options = (
+        application_data["options"]
+        if "options" in application_data
+        else Options(
+            target=application_data["targets"],
+            input=[application_data["path"]],
+            cli=config["windup_cli_path"],
+            source_mode=True,
+            output=f"{application_config['plugin_cache_path']}/{uuid}",
+        )
+    )
+
+    configuration = Configuration(name=app_name, id=uuid, options=options)
+
+    configurations.configurations.append(configuration)
+
+    # endregion
+
+    # convert the object to json and write to the model.json file
+    final_configuration_json = json.dumps(configurations.to_dict())
+    write_data_to_file(model_json_path, final_configuration_json)
+
+    yield configurations
+
+    # Construct default configuration object
```

Comment:
I understand the purpose of this, but could you clarify in what scenario these lines get executed ? We already have an if block . Can this scenario be covered there ?

Also, 'configurations' is not being yielded here .

Reply:
@midays . The fixture tear down looks good now . Thanks for updating the PR.